### PR TITLE
Remove `-Z two-phase-borrows` and `-Z two-phase-beyond-autoref`

### LINKED ISSUE #60331: Remove `-Ztwo-phase-borrows`
This flag does nothing, so can be removed from `librustc/session/config.rs`.

The `-Z two-phase-beyond-autoref` flag could also be removed, since it will usually result in an ICE rather than anything helpful. Removing this will require a bit more work to remove any usage of the flag.

### PR DESCRIPTION
**Pull request by JohnTitor**
fixes #60331

### CODE REVIEW

File: src/librustc_mir/borrow_check/borrow_set.rs [hunk 1]
```diff
@@ -304,7 +304,6 @@ impl<'a, 'gcx, 'tcx> GatherBorrows<'a, 'gcx, 'tcx> {
     /// Activation phases.
     fn allow_two_phase_borrow(&self, kind: mir::BorrowKind) -> bool {
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/borrow_set.rs]: Could you remove this function and replace uses of it with the one in `path_utils`?

Comment (JohnTitor) [reply src/librustc_mir/borrow_check/borrow_set.rs]: @matthewjasper Thanks for reviewing! I removed the function.

**Pull request merged**